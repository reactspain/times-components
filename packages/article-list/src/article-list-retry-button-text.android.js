import React from 'react';
import { Text } from "react-native";
import styles from "./styles";

const RetryButtonText = () => (
  <Text style={styles.listErrorButtonText}>RETRY</Text>
);

export default RetryButtonText;