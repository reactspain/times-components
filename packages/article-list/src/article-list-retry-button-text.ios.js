import React from 'react';
import { Text } from "react-native";
import styles from "./styles";

const RetryButtonText = () => (
  <Text style={styles.listErrorButtonText}>Retry</Text>
);

export default RetryButtonText;