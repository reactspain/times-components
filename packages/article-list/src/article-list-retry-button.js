import React from "react";
import { Text, TouchableOpacity, View } from "react-native";
import propTypes from "./article-list-retry-button-prop-types";
import RetryButtonText from "./article-list-retry-button-text";
import styles from "./styles";

const ArticleListRetryButton = ({ refetch }) => (
  <TouchableOpacity accessible accessibilityLabel="Retry" onPress={refetch}>
    <View style={styles.listErrorButton}>
      <RetryButtonText />
    </View>
  </TouchableOpacity>
);

ArticleListRetryButton.propTypes = propTypes;

export default ArticleListRetryButton;
