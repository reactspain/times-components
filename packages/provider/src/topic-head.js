import gql from "graphql-tag";
import connectGraphql from "./connect";

export const query = gql`
  query TopicQuery($slug: Slug!) {
    topic(slug: $slug) {
      name
      description
    }
  }
`;

const propsToVariables = ({
  slug
}) => ({
  slug
});

export default connectGraphql(query, propsToVariables);
