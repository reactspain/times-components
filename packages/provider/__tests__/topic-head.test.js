import React from "react";
import renderer from "react-test-renderer";
import { MockedProvider } from "@times-components/utils";
import { fixtureGenerator } from "@times-components/provider-test-tools";
import { TopicHeadProvider } from "../src/provider";

const pageSize = 5;

const mocks = fixtureGenerator.makeTopicHeadMocks({ delay: 0 });

const renderComponent = child =>
  renderer.create(
    <MockedProvider mocks={mocks}>
      <TopicHeadProvider debounceTimeMs={0} slug="chelsea">
        {child}
      </TopicHeadProvider>
    </MockedProvider>
  );

describe("TopicHeadProvider", () => {
  it("returns query result", done => {
    renderComponent(({ isLoading, topic }) => {
      if (!isLoading) {
        expect(topic).toMatchSnapshot();
        done();
      }

      return null;
    });
  });
});
