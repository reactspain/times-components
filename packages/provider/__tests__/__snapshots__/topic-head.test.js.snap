// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopicHeadProvider returns query result 1`] = `undefined`;
